health/doc.go

```diff
@@ -0,0 +1,143 @@
+// Package health provides a generic health checking framework. The health
+// package works expvar style. By importing the package the debug server is
+// getting a `/debug/health` endpoint that returns the current status of the
+// application. If there are no errors, `/debug/health` will return a HTTP 200
+// status, together with an empty JSON reply `{}`. If there are any checks
+// with errors, the JSON reply will include all the failed checks, and the
+// response will be have an HTTP 500 status.
+//
+// A Check can either be run synchronously, or asynchronously. We recommend
+// that most checks are registered as an asynchronous check, so a call to the
+// `/debug/health` endpoint always returns immediately. This pattern is
+// particularly useful for checks that verify upstream connectivity or
+// database status, since they might take a long time to return/timeout.
+//
+// ## Installing
+//
+// To install health, just import it in your application:
+//
+// `import "github.com/docker/distribution/health"`
+//
+// You can also (optionally) import `health/api` that will add two convenience
+// endpoints: `/debug/health/down` and `/debug/health/up`. These endpoints add
+// "manual" checks that allow the service to quickly be brought in/out of
+// rotation.
+//
+// `import _ "github.com/docker/distribution/registry/health/api"`
+//
+// ```bash
```

Comment:
Please check the rendering of these doc comments by running `godoc -http :8080` and then navigating to the package (should be under something like `localhost:8080/pkg/github.com/docker/distribution/health`). I am not sure that the full markdown formatting will work correctly.